### PR TITLE
Sandbox app reorganisation

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
### Summary

I started working on some tests for #980 but I wanted to propose this little refactor first...

Moves all files related to the dummy app (mounted in the test environment) from `/test` to `/test/dummy`. This makes it a lot clearer which files are related to tests and which are part of the dummy app.

What do you think?

### Other Information

It's easier to grok the changes with a visual example:

**Before**

Screenshot from 2021-06-26 11-40-34

**After**

Screenshot from 2021-06-26 11-43-08

### CODE REVIEW

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ title: Changelog
 
 ## main
 
+* Moved dummy app files to /test/dummy
```
Comment (joelhawksley) [inline docs/CHANGELOG.md:10]:
```suggestion
* Move dummy app files to /test/dummy.
```

**Review by joelhawksley — COMMENTED** [review]
@Matt-Yorkley thanks for making this change!

If it's not too much hassle, would you be willing to rename the application from `dummy` to `example` or something similar in this PR? In some contexts `dummy` can be perceived as less than friendly and inclusive ❤️

**Comment by Matt-Yorkley** [comment]
> In some contexts `dummy` can be perceived as less than friendly and inclusive

Ah! I wasn't aware of that.

> `example` or something similar

I mostly use RSpec, where the word "example" often refers to tests themselves. Is there some conceptual overlap / blurryness there? Maybe not so much with Minitest? 

If it's purpose is to simulate a downstream consumer that's including the engine in itself, maybe `consumer` conveys it's role well?

Naming things is hard :sweat_smile:

**Comment by Spone** [comment]
> Naming things is hard sweat_smile

Some ideas: `sample`, `mockup`?

**Comment by joelhawksley** [comment]
> > Naming things is hard
> 
> Some ideas: `sample`, `mockup`, `sandbox`?

I'm 👍🏻 to `sample` and `sandbox`. Pick one of those?

**Comment by Matt-Yorkley** [comment]
Updated to `sandbox` :ok_hand:

**Review by joelhawksley — APPROVED** [review]

**Pull request merged**